packages/react/src/components/Alert/Alert.tsx

```diff
@@ -0,0 +1,138 @@
+} from '../../lib'
+import { RenderResultConfig } from 'src/lib/renderComponent'
+import { defaultBehavior } from '../../lib/accessibility'
+import { Accessibility } from '../../lib/accessibility/types'
+import { ComponentEventHandler, ReactProps, ShorthandValue } from '../../types'
+import Box from '../Box/Box'
+import Button, { ButtonProps } from '../Button/Button'
+
+export interface AlertSlotClassNames {
+  content: string
+  closeButton: string
+}
+
+export interface AlertProps extends UIComponentProps, ContentComponentProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   * @default defaultBehavior
+   */
+  accessibility?: Accessibility
+
+  /** Controls Alert's relation to neighboring items. */
+  attached?: boolean | 'top' | 'bottom'
+
+  /** Button used for dismissing the Alert. */
+  closeButton?: ShorthandValue<ButtonProps>
+
+  /** An alert may be formatted to display a danger message. */
+  danger?: boolean
+
+  /** An alert may be formatted to display information. */
+  info?: boolean
+
+  /**
+   * The callback used when the user chooses to hide the Alert.
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onClose?: ComponentEventHandler<AlertProps>
+
+  /** An alert may be formatted to display a successful message. */
+  success?: boolean
+
+  /** An alert may be formatted to display a warning message. */
+  warning?: boolean
+}
+
+/**
+ * A Alert displays information that explains nearby content.
+ */
+class Alert extends UIComponent<ReactProps<AlertProps>> {
+  static displayName = 'Alert'
+  static className = 'ui-alert'
+
+  static slotClassNames: AlertSlotClassNames = {
+    content: `${Alert.className}__content`,
+    closeButton: `${Alert.className}__closeButton`,
+  }
+
+  static propTypes = {
+    ...commonPropTypes.createCommon(),
+    attached: PropTypes.oneOfType([PropTypes.bool, PropTypes.oneOf(['top', 'bottom'])]),
+    closeButton: customPropTypes.itemShorthand,
+    danger: PropTypes.bool,
+    info: PropTypes.bool,
+    onClose: PropTypes.func,
+    success: PropTypes.bool,
+    warning: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    accessibility: defaultBehavior,
+    closeButton: { icon: 'close' },
+  }
+
+  handleCloseButtonOverrides = (predefinedProps: ButtonProps) => ({
+    onClick: (e: React.SyntheticEvent, buttonProps: ButtonProps) => {
+      this.handleOnClose(e)
+      _.invoke(predefinedProps, 'onClick', e, buttonProps)
+    },
+  })
+
+  handleOnClose = (e: React.SyntheticEvent) => {
+    _.invoke(this.props, 'onClose', e, this.props)
+  }
+
+  renderContent = ({ styles }: RenderResultConfig<AlertProps>) => {
+    const { closeButton, content, onClose } = this.props
+    return (
+      <>
+        {Box.create(content, {
+          defaultProps: {
+            className: Alert.slotClassNames.content,
+            styles: styles.content,
+          },
+        })}
+        {onClose &&
```

Comment:
I don't like that we are using the `onClose` prop here as a decision of whether we want to render the `closeButton`. The user may provide just the `closeButton`, and add `onClick` event there, but still the button will not be shown. We may revisit this logic in a follow up PR.

Reply:
good point... I was thinking about it but wanted to avoid adding one more prop; the alternative would be to add a `closable` prop and do this:
```jsx
<Alert content="Close me!" closable onClose={() => /* code for closing */} />
```

but then we might end up with the following scenario if the user does not provide `onClose` or `onClick` on the `closeButton` slot:

```jsx
<Alert content="Close me!" closable />
```

**User clicks on the icon but it doesn't close because the component does not handle its visibility state**. At least in the current implementation it's a bit more obvious that the user needs write some code in the `onClose` handler for the close to happen.

## Proposed solution for v2:

Probably it'd be better to make this component autocontrolled; we can have there an internal `open` flag exported as an auto-controlled prop for the visibility of the `Alert`. In that case, the scenario above will be solved by doing:

- uncontrolled mode => the `Alert` will close because of internal implementation (similar to click outside in a popup):
```jsx
<Alert content="Close me!" closable />
```

- or controlled mode => the `Alert` will close because of the parent component code:
```jsx
class AlertExampleClosable extends React.Component<{}, { open: boolean }> {
  state = { open: true }

  render() {
    return (
      this.state.open && (
        <Alert
          open={this.state.open}
          content="Close me"
          onClose={() => this.setState({ open: false })}
        />
      )
    )
  }
}
```

@mnajdova @layershifter what do you guys think?

Reply:
Makes sense!

Reply:
My vote of ACC 👍

Reply:
added to the RFC #1061 - I will create one for follow-up (Alert v2 or something)